.github/workflows/release.yaml

```diff
@@ -0,0 +1,100 @@
+      branch:
+        description: 'Branch to Release'
+        required: false
+        default: 'main'
+permissions:
+  contents: write
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Update Sources
+        run: |
+          mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${{ github.event.inputs.releaseVersion }}
+          mvn -B clean verify
+          sed --in-place --regexp-extended "s|flink-examples-data-generator:([^[:space:]]*)|flink-examples-data-generator:${{ github.event.inputs.releaseVersion }}|g" recommendation-app/data-generator.yaml
+          sed --in-place --regexp-extended "s|flink-examples-data-generator:([^[:space:]]*)|flink-examples-data-generator:${{ github.event.inputs.releaseVersion }}|g" interactive-etl/data-generator.yaml
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+          git add .
+          git commit -m "Release ${{ github.event.inputs.releaseVersion }}"
+          git tag -m "v${{ github.event.inputs.releaseVersion }}" -a "v${{ github.event.inputs.releaseVersion }}"
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Test Build Image
+        uses: docker/build-push-action@v6
+        with:
+          context: data-generator/
+          platforms: linux/amd64,linux/arm64
+          push: false
+          file: data-generator/Dockerfile
+      
+      - name: Push Tag
+        run: |
+          git push
+          git push origin "v${{ github.event.inputs.releaseVersion }}"
+ 
+      - name: Image metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ secrets.IMAGE_REPO_HOSTNAME }}/${{ secrets.IMAGE_REPO_NAMESPACE }}/flink-examples-data-generator
+          tags: |
+            type=raw,value=v${{ github.event.inputs.releaseVersion }}
+            type=raw,value=latest
+ 
+      - name: Login to Quay
+        uses: docker/login-action@v3
+        with:
+          registry: "${{ secrets.IMAGE_REPO_HOSTNAME }}"
+          username: "${{ secrets.IMAGE_REPO_USERNAME }}"
+          password: "${{ secrets.IMAGE_REPO_PASSWORD }}"
+
+      - name: Build and Push Image
+        uses: docker/build-push-action@v6
+        with:
+          context: data-generator/
+          platforms: linux/amd64,linux/arm64
+          push: true
+          file: data-generator/Dockerfile
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
similar to whats in the integration.yaml, I think it's okay for a start though. pushing to main and pushing the tag using the default GIT_TOKEN do not trigger the other workflow. We'd need to use another PAT stored in a secret, which is a bit crap.